### ISSUE TITLE
I18n localisation files do not support subdirectories.

### ISSUE THREAD

**Issue by lux-shaun** [bridgetownrb/bridgetown#598]
Locale files have to be directly in src/_locales to work (e.g. src/_locales/en.yml), which limits the ability to organise multiple files. 

It looks like changing the load path in localizable.rb from 
```
Dir["#{in_source_dir("_locales")}/*.{json,rb,yml}"].each do |locale_path|
  I18n.load_path << locale_path
end
```
to
```
Dir["#{in_source_dir("_locales")}/**/*.{json,rb,yml}"].each do |locale_path|
  I18n.load_path << locale_path
end
```
would resolve this, or even one-lining it as Rails does with `I18n.load_path += Dir["#{in_source_dir("_locales")}/**/*.{json,rb,yml}"]`.

**Comment by jaredcwhite** [comment]
@robodinoshaun That makes sense to me. Would you be willing to submit a PR with the proposed fix (I'm fine with either option)?

**Comment by robodinoshaun** [comment]
Sure, I should be able to get around to that soon.

A loosely related suggestion would be updating the fallback chain for regional language variants to include the base locale (e.g. fr-CN -> fr -> en rather than simply fr-CN -> en)

**Comment by jaredcwhite** [comment]
@robodinoshaun That seems reasonable. Out of curiosity, is that something Rails does automatically? Or is it a manual config step?

**Comment by robodinoshaun** [comment]
@jaredcwhite yeah, it does.

**Comment by robodinoshaun** [comment]
@jaredcwhite Actually, it's a function of ruby-i18n itself.
https://github.com/ruby-i18n/i18n/blob/master/lib/i18n/locale/fallbacks.rb

```
irb(main):001:0> require 'i18n'
irb(main):003:0> require 'i18n/backend/fallbacks'
irb(main):007:0> I18n::Backend::Simple.send(:include, I18n::Backend::Fallbacks)
=> I18n::Backend::Simple
irb(main):010:0> I18n.fallbacks['es-ES']
=> [:"es-ES", :es]
```

This doesn't include the default locale, but Rails has a config setting that ultimately sets `I18n.fallbacks = [I18n.default_locale]`, which includes it at the end of the chain.  Demonstrated here:

```
irb(main):003:0> I18n::Backend::Simple.send(:include, I18n::Backend::Fallbacks)
=> I18n::Backend::Simple
irb(main):004:0> I18n.fallbacks['fr-CA']
=> [:"fr-CA", :fr]
irb(main):005:0> I18n.fallbacks = [I18n.default_locale]
=> [:en]
irb(main):006:0> I18n.fallbacks['fr-CA']
=> [:"fr-CA", :fr, :en]
```

**Comment by jaredcwhite** [comment]
@robodinoshaun Interesting! I'll have to take a second look at this along with #607…the sort of wonky looking code that was in there for fallbacks I had added for a reason, but now I can't for the life of me remember the reason! (should have added a comment!)

**Comment by robodinoshaun** [comment]
Interesting.  If there IS a reason it was necessary, I would be curious as to what it is.

In which case I guess the required change would be to add the top-level directly to code as written.
I considered this initially but when I realised how the I18n library handled it I felt it was a better.

```
        I18n.fallbacks = (config[:available_locales] + [:en]).uniq.to_h do |available_locale|
          [available_locale, [available_locale, available_locale.to_s.first(2).to_sym, locale, :en].uniq]
        end
```

The tests in the PR should still prove useful either way, I think.

**Comment by sandstrom** [comment]
Something like this should work:

```ruby
# plugins/builders/custom_setup.rb
class Builders::CustomSetup < SiteBuilder
  def build
    I18n.load_path << Dir["#{site.source}/_locales/**/*.yml"]
  end
end
```

**Comment by jaredcwhite** [comment]
I think this is addressed now in v1.2.

**Comment by sandstrom** [comment]
Awesome! @jaredcwhite
